terminus/generators/monolane_generator.py

```diff
@@ -0,0 +1,203 @@
+from __future__ import print_function
+
+from collections import OrderedDict
+from math import atan2
+from math import degrees
+import yaml
+
+from .city_visitor import CityVisitor
+from models.intersection_node import IntersectionNode
+
+
+class MonolaneGenerator(CityVisitor):
+    def start_city(self, city):
+        """Called when a visitation of a new city starts."""
+        # Reset all of the class variables used until stop_city is called.
+        self.monolane = {
+            'maliput_monolane_builder': OrderedDict([
+                ('id', city.name),
+                ('lane_bounds', [-2, 2]),
+                ('driveable_bounds', [-4, 4]),
+                ('position_precision', 0.01),
+                ('orientation_precision', 0.5),
+                ('points', OrderedDict()),
+                ('connections', OrderedDict()),
+                ('groups', OrderedDict()),
+            ])
+        }
+        self.streets = []
+        self.point_name_by_street_node_dict = {}
+        self.intersection_index = 0
+        self.intersection_index_by_node = {}
+
+    def node_ident(self, street, node):
```

Comment:
I have to admit, I found the fact that a `CityVisitor` ran another `CityVisitor` to be very confusing and was one of the reasons I avoided copying the rndf generator directly. It seemed unnecessarily complicated. I don't mind refactoring the the storage and methods for calculating the identity to another class, but I'd rather not make it a `CityVisitor` like `RNDFIdMapper` unless you really want it that way.

It's not clear to me if this can be generalized or not, as each format may have different rules for what may go into a name, but you might be able to generalize the interface. I'll look and see where there is overlap with what I am doing and what is in the interface for the other `*IdMapper` classes.